documentation/writing_tests.md

```diff
@@ -16,4 +16,29 @@
 
 ## Structure of tests
 
-## Individual tests
\ No newline at end of file
+## Writing test cases
+
+a. On titles
+	i. Test should only test one thing. The titles should reflect that.
+b. Each test should just test one function or method.
```

Comment:
this is tricky to enforce for stateful things, like clients. I like the spirit of it, but it wouldn't make sense for operation tests as much as it does for unit tests of an internal helper or something

Reply:
I have updated this PR with more concrete information. I will come back tomorrow to go through  all of my notes and see if I'm missing anything from the feedback I got on the meetings I had through the past weeks.